File: wc115/case1/case1.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isCompleteTree(root *TreeNode) bool {
	var stack []*TreeNode
	var ns []*TreeNode
	var foundnull bool

	stack = append(stack, root)

	for len(stack) > 0 {
		for _, v := range stack {
			fmt.Println(v)
			if v != nil {
				if foundnull == true {
					return false
				}
				if v.Left != nil {
					ns = append(ns, v.Left)
				} else {
					ns = append(ns, nil)
				}
				if v.Right != nil {
					ns = append(ns, v.Right)
				} else {
					ns = append(ns, nil)
				}
			} else {
				if foundnull == true {
					continue
				} else {
					foundnull = true
				}
			}
		}
		stack = ns
		ns = []*TreeNode{}
	}
	return true
}

func main() {
	n := &TreeNode{
		Val: 2,
		Left: &TreeNode{
			Val: 3,
		},
		Right: &TreeNode{
			Val: 4,
		},
	}
	fmt.Println(isCompleteTree(n))
}

File: wc124/case1.cpp
#include <iostream>
#include <vector>

using namespace std;

//  Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    bool isCousins(TreeNode* root, int x, int y) {
        auto p1 = _bfs(root, nullptr, 0, x);
        auto p2 = _bfs(root, nullptr, 0, y);

        if (p1.first == p2.first && p1.second != p2.second) {
            return true;
        }
        return false;
    }
    // pair <depth, parent>
    pair<int, int> _bfs(TreeNode* cur, TreeNode* p, int depth, int t) {
        ++depth;
        auto res = make_pair(depth, -1);
        if (cur->val == t) {
            if (p != nullptr) {
                res.second = p->val;
            }
            return res;
        } else {
            if (cur->left != nullptr) {
                res = _bfs(cur->left, cur, depth, t);
                if (res.second != -1) {
                    return res;
                }
            }
            if (cur->right != nullptr) {
                res = _bfs(cur->right, cur, depth, t);
                if (res.second != -1) {
                    return res;
                }
            }
        }
        return res;
    }
};

int main() {}